disnake/file.py

```diff
@@ -111,3 +112,37 @@ def close(self) -> None:
         self.fp.close = self._closer
         if self._owner:
             self._closer()
+
+    def __sizeof__(self) -> int:
+        return sys.getsizeof(self.fp)
+
+    @property
+    def closed(self) -> bool:
+        """:class:`bool`: Wheter a file or bytes object is closed.
+
+        This is a shorthand for ``File.fp.closed``.
+
+        .. versionadded:: 2.8
+        """
+        return self.fp.closed
+
+    @property
+    def bytes_length(self) -> int:
+        """:class:`int`: The bytes length of the :attr:`~File.fp` object.
+
+        .. versionadded:: 2.8
+        """
+        if self.fp.tell() != 0:
+            self.reset()
```

Comment:
The file may have already been partially read at this point; instead of resetting to `self._original_pos`, this should save the current position and reset back to that later, to avoid weird side effects.

```py
In [3]: file = File(BytesIO(b'a' * 1000))

In [4]: file.fp.read(16)
Out[4]: b'aaaaaaaaaaaaaaaa'

In [5]: file.fp.tell()
Out[5]: 16

In [6]: file.bytes_length
Out[6]: 1000

In [7]: file.fp.tell()
Out[7]: 0
```

Something like this would work: https://github.com/aio-libs/aiohttp/blob/8e92c4c7c305f914c17f41539da20822ea7baddf/aiohttp/payload.py#L367-L370